cluster/src/main/scala/io/snappydata/gemxd/SparkSQLExecuteImpl.scala

```diff
@@ -482,6 +482,7 @@ object SnappySessionPerConnection {
     if (session != null) session
     else {
       val session = SnappyContext().snappySession
```

Comment:
For corner cases, this should better check that SparkContext is not set instead of throwing NPE like:
```scala
val session = SparkContext.globalSparkContext match {
   // use a CancelException to force failover by client to another lead if available
  case null => throw new CacheClosedException("No SparkContext ...")
  case sc => new SnappySession(sc)
}
```